sqlglot/parser.py

```diff
@@ -5475,6 +5484,44 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> t.Optional[exp.Heredoc]:
+        tags = []
+        tag_text = None
+
+        if self._match_text_seq("$"):
+            tags.append("$")
+        else:
+            return None
+
+        if self._curr and self._is_connected():
```

Comment:
I think `self._curr` isn't needed here? Just `_is_connected()` seems to be enough.